### DISCUSSION TITLE
Strawberry/GraphQL filter syntax porting in Netbox 4.0

### DISCUSSION THREAD

**Discussion by ee-github** [netbox-community/netbox#17236] [Help Wanted!]
Hi All,

Can anyone suggest the correct filter syntax in the new GraphQL implementation to filter devices by checking if their name contains any of a list of strings (possible substrings of the device name)?  This works in Netbox 3.0 with the following query and variables:

```
query Devices($names:[String]) {
  device_list(name__ic: $names) {
    name
    status
  }
}

{
  "names": ["switch-x", "router-a"]
}
```
The above returns data for "switch-x1, switch-x2, router-a5..."

What would be the equivalent in the new Strawberry implementation?

The `in_list` operator seems to only match on exact strings.  So this doesn't work:

```
query Devices($names:[String!]) {
  device_list(filters: {name: {in_list: $names}}) {
    name
    status
  }
}
```

And the `i_contains` operator only seems to take a single string, not a list.

Is anyone aware of any way to chain these two together, or another solution?

Thanks in advance.  Any help appreciated.

**Comment by ee-github** [comment]
To add:  I suppose I can use `i_regex` but then I would have to first construct the logical ORed string, which seems messy.  Hopefully someone has a cleaner idea.

**Comment by ProtonBruno** [comment]
Has anyone been able to move fw on this issue?
We are also looking for a way to do this, but without success so far...
